src/argilla_sdk/datasets/_export.py

```diff
@@ -0,0 +1,184 @@
+        if self.exists():
+            self.records.to_json(path=records_path)
+
+        return path
+
+    @classmethod
+    def from_disk(
+        cls,
+        path: str,
+        target_workspace: Optional[Union["Workspace", str]] = None,
+        target_name: Optional[str] = None,
+        client: Optional["Argilla"] = None,
+    ) -> "Dataset":
+        """Imports a dataset from disk as a directory containing the dataset model, settings and records.
+        The directory should be defined using the `to_disk` method.
+
+        Parameters:
+        path (str): The path to the directory containing the dataset model, settings and records.
+        target_workspace (Union[Workspace, str], optional): The workspace to import the dataset to. Defaults to None and default workspace is used.
+        target_name (str, optional): The name to assign to the new dataset. Defaults to None and the dataset's source name is used, unless it already exists, in which case a unique UUID is appended.
+        client (Argilla, optional): The client to use for the import. Defaults to None and the default client is used.
+        """
+
+        client = client or Argilla._get_default()
+
+        dataset_path, settings_path, records_path = cls._define_child_paths(path=path)
+
+        logging.info(f"Loading dataset from {dataset_path}")
+        logging.info(f"Loading settings from {settings_path}")
+        logging.info(f"Loading records from {records_path}")
+
+        if not os.path.exists(dataset_path):
+            raise FileNotFoundError(f"Dataset model not found at {dataset_path}")
+
+        dataset_model = cls._load_dataset_model(path=dataset_path)
+
+        # Get the relevant workspace_id of the incoming dataset
+        if isinstance(target_workspace, str):
+            workspace_id = client.workspaces(target_workspace).id
+        elif isinstance(target_workspace, Workspace):
+            workspace_id = target_workspace.id
+        else:
+            warnings.warn("Workspace not provided. Using default workspace.")
+            workspace_id = client.workspaces.default.id
+        dataset_model.workspace_id = workspace_id
+
+        # Get a relevant and uniwue name for the incoming dataset.
+        if target_name:
+            logging.warning(f"Changing dataset name from {dataset_model.name} to {target_name}")
+            dataset_model.name = target_name
+        elif client.api.datasets.name_exists(name=dataset_model.name, workspace_id=workspace_id):
+            logging.warning(f"Loaded dataset name {dataset_model.name} already exists. Changing to unique UUID.")
+            dataset_model.name = f"{dataset_model.name}_{uuid4()}"
+
+        # Create the dataset and load the settings and records
+        dataset = cls.from_model(model=dataset_model, client=client)
+        dataset.settings = Settings.from_json(path=settings_path)
+        if os.path.exists(records_path):
+            dataset.records.from_json(path=records_path)
```

Comment:
At some point here, the dataset should be created, right?  but, by reading this code block is not clear if this is done somewhere.